controllers/dscinitialization/dscinitialization_controller.go

```diff
@@ -115,6 +119,49 @@ func (r *DSCInitializationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		return reconcile.Result{}, err
 	}
 
+	// Cleanup old namespace's resource, but keep old namespace
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	s, _ := labels.Parse("opendatahub.io/generated-namespace=true")
+	ODHNamespaceList, err := clientset.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{
+		LabelSelector: s.String(),
+	})
+	if !apierrs.IsNotFound(err) {
+		return reconcile.Result{}, err
+	}
+
+	resourceTypes := []client.Object{
+		&appsv1.Deployment{},
+		&corev1.Service{},
+		&corev1.Secret{},
+		&corev1.ConfigMap{},
+		&netv1.NetworkPolicy{},
+		&authv1.Role{},
+		&authv1.RoleBinding{},
+		&authv1.ClusterRole{},
+		&authv1.ClusterRoleBinding{},
+	}
+
+	for _, matchedNamespace := range ODHNamespaceList.Items {
+		if matchedNamespace.Name != namespace && matchedNamespace.Name != instance.Spec.Monitoring.Namespace {
+			for _, obj := range resourceTypes {
+				if err = r.Client.DeleteAllOf(ctx, obj, client.InNamespace(matchedNamespace.Name), client.MatchingLabels{"opendatahub.io/component": "true"}); err != nil {
+					r.Log.Error(err, "Failed to cleanup old namespace left-over resource.", "Old Namespace", matchedNamespace.Name, "Type of Resource", obj.GetName())
+				}
+			}
+		}
+	}
+	if namespace != "opendatahub" { // opendatahub does not have "opendatahub.io/generated-namespace=true" on it
```

Comment:
What if a user updates a namespace multiple times?

For e.g updates from `opendatahub`--> `customns1` ---> `customns2`

Reply:
so, it will first go thro the namespace match label selector part. 
when "customns1" was created it gets label, 

` if namespace != "opendatahub"` is only used because in the existing cluster, `opendatahub`ns  might not have the label applied to it.